File: src/cshape_code/DesignPattern/SingletonPattern/SingletonObjectE.cs
using System;
using System.Threading;
namespace DesignPattern.SingletonPattern
{
    /// <summary>
    /// Singleton object e.
    /// Lazy: true
    /// Thread safe: true
    /// </summary>
    public class SingletonObjectE
    {
        public static int count;

        private SingletonObjectE()
        {
            Interlocked.Increment(ref count);
        }

        private static class SingletonObjectHolder
        {
            public static SingletonObjectE singletonObjectE = new SingletonObjectE();
        }

        public static SingletonObjectE GetSingletonObjectE()
        {
            return SingletonObjectHolder.singletonObjectE;
        }

        public void ShowMessage()
        {
            Console.WriteLine("Singleton Object B");
        }
    }
}

File: src/java_code/singletonpattern/SingletonObjectE.java
package singletonpattern;

public class SingletonObjectE{
    private static class SingletonObjectEHolder{
        public static SingletonObjectE singletonObjectE = new SingletonObjectE();
    }

    private SingletonObjectE(){}

    public static SingletonObjectE getSingletonObjectE(){
        return SingletonObjectEHolder.singletonObjectE;
    }
}